commit 3e23f36a3c17863b11d9311af204e486e63cac19
Author: Jussi Pakkanen <jpakkane@gmail.com>
Date:   Thu Mar 29 23:32:17 2018 +0300

    Fix detection to work with newer Java compilers.

diff --git a/mesonbuild/environment.py b/mesonbuild/environment.py
index ff7c706e..0115fb35 100644
--- a/mesonbuild/environment.py
+++ b/mesonbuild/environment.py
@@ -674,7 +674,7 @@ class Environment:
         except OSError:
             raise EnvironmentException('Could not execute Java compiler "%s"' % ' '.join(exelist))
         version = search_version(err)
-        if 'javac' in err:
+        if 'javac' in out or 'javac' in err:
             return JavaCompiler(exelist, version)
         raise EnvironmentException('Unknown compiler "' + ' '.join(exelist) + '"')
 
