commit 3e12af383a81f88838918f1b1cd217a09b22597c
Author: Jussi Pakkanen <jpakkane@gmail.com>
Date:   Mon Apr 2 12:38:15 2018 +0300

    Imported Debian fixes by Matthias Klose.

diff --git a/cross/ubuntu-armhf.txt b/cross/ubuntu-armhf.txt
index 6246ffe0..fec8ce7d 100644
--- a/cross/ubuntu-armhf.txt
+++ b/cross/ubuntu-armhf.txt
@@ -1,8 +1,8 @@
 [binaries]
 # we could set exe_wrapper = qemu-arm-static but to test the case
 # when cross compiled binaries can't be run we don't do that
-c = '/usr/bin/arm-linux-gnueabihf-gcc-7'
-cpp = '/usr/bin/arm-linux-gnueabihf-g++-7'
+c = '/usr/bin/arm-linux-gnueabihf-gcc'
+cpp = '/usr/bin/arm-linux-gnueabihf-g++'
 rust = ['rustc', '--target', 'arm-unknown-linux-gnueabihf', '-C', 'linker=/usr/bin/arm-linux-gnueabihf-gcc-7']
 ar = '/usr/arm-linux-gnueabihf/bin/ar'
 strip = '/usr/arm-linux-gnueabihf/bin/strip'
diff --git a/run_tests.py b/run_tests.py
index 1cc39833..648e6ce1 100755
--- a/run_tests.py
+++ b/run_tests.py
@@ -131,7 +131,7 @@ def get_fake_options(prefix):
     return opts
 
 def should_run_linux_cross_tests():
-    return shutil.which('arm-linux-gnueabihf-gcc-7') and not platform.machine().lower().startswith('arm')
+    return shutil.which('arm-linux-gnueabihf-gcc') and not platform.machine().lower().startswith('arm')
 
 def run_configure_inprocess(meson_command, commandlist):
     old_stdout = sys.stdout
